apps/api/src/server/applications/application/__tests__/start-case.test.js

```diff
@@ -0,0 +1,130 @@
+	description: null,
+	subSectorId: null,
+	zoomLevelId: null,
+	regions: []
+});
+const applicationInPreApplicationState = applicationFactoryForTests({
+	id: 4,
+	status: 'pre_application'
+});
+
+const validationCaseDetailsInclusions = {
+	ApplicationDetails: { include: { subSector: false, zoomLevel: false, regions: true } },
+	CaseStatus: { where: { valid: true } }
+};
+
+findUniqueStub.withArgs({ where: { id: 1 } }).returns(applicationReadyToStart);
+findUniqueStub
+	.withArgs({ where: { id: 1 }, include: validationCaseDetailsInclusions })
+	.returns(applicationReadyToStart);
+findUniqueStub.withArgs({ where: { id: 2 } }).returns(null);
+findUniqueStub.withArgs({ where: { id: 3 } }).returns(applicationWithMissingInformation);
+findUniqueStub
+	.withArgs({ where: { id: 3 }, include: validationCaseDetailsInclusions })
+	.returns(applicationWithMissingInformation);
+findUniqueStub.withArgs({ where: { id: 4 } }).returns(applicationInPreApplicationState);
+findUniqueStub
+	.withArgs({ where: { id: 4 }, include: validationCaseDetailsInclusions })
+	.returns(applicationInPreApplicationState);
+
+const updateStub = sinon.stub();
+const updateManyCaseStatusStub = sinon.stub();
+const createCaseStatusStub = sinon.stub();
+
+test.before('set up mocks', () => {
+	sinon.stub(databaseConnector, 'case').get(() => {
+		return { findUnique: findUniqueStub, update: updateStub };
+	});
+
+	sinon.stub(databaseConnector, 'caseStatus').get(() => {
+		return { updateMany: updateManyCaseStatusStub, create: createCaseStatusStub };
+	});
+
+	sinon.stub(databaseConnector, 'regionsOnApplicationDetails').get(() => {
+		return { deleteMany: sinon.stub() };
+	});
+
+	sinon.stub(Prisma.PrismaClient.prototype, '$transaction');
+	sinon.useFakeTimers({ now: 1_649_319_144_000 });
+});
+
+test('starts application if all needed information is present', async (t) => {
+	const response = await request.post('/applications/1/start');
+
+	t.is(response.status, 200);
+	t.deepEqual(response.body, {
+		id: 1,
+		reference: 'some new reference',
+		status: 'Pre-Application'
+	});
+
+	sinon.assert.calledWith(updateManyCaseStatusStub, {
+		where: { id: { in: [1] } },
+		data: { valid: false }
+	});
+
+	sinon.assert.calledWith(createCaseStatusStub, { data: { status: 'pre_application', caseId: 1 } });
+
+	sinon.assert.calledWith(updateStub, {
+		where: { id: 1 },
+		data: { modifiedAt: new Date(1_649_319_144_000) },
+		include: { serviceCustomer: true }
+	});
+});
+
+test('throws an error if the application id is not recognised', async (t) => {
+	const response = await request.post('/applications/2/start');
+
+	t.is(response.status, 400);
+	t.deepEqual(response.body, {
+		errors: {
+			id: 'Must be an existing application'
+		}
+	});
+});
+
+test('throws an error if the application does not have all the required information to start', async (t) => {
+	const response = await request.post('/applications/3/start');
+
+	t.is(response.status, 400);
+	t.deepEqual(response.body, {
+		error: {
```

Comment:
is it not 'errorS'?

Reply:
So at the minute, the default error handler returns `error` not `errors`. And I was not going to change it until I was sure that it was ok on the Web side. Do you know if it might be an issue?

Reply:
it's not an issue as long as it's always 'error'. I can see a few lines up and and a few lines down you're returning errors

Reply:
So validation error handler returns `errors`, hence the difference. i can make it so it is consistent?

Reply:
yes, that would be better

Reply:
done